crates/fuel-core/src/graphql_api/required_fuel_block_height_extension.rs

```diff
@@ -0,0 +1,112 @@
+use async_graphql::{
+    extensions::{
+        Extension,
+        ExtensionContext,
+        ExtensionFactory,
+        NextPrepareRequest,
+    },
+    Pos,
+    Request,
+    ServerError,
+    ServerResult,
+};
+use fuel_core_types::fuel_types::BlockHeight;
+use std::{
+    any::TypeId,
+    sync::Arc,
+};
+use tokio::sync::Mutex;
+
+use super::{
+    api_service::RequiredHeight,
+    database::ReadView,
+};
+
+/// The extension that implements the logic for checking whether
+/// the precondition that REQUIRED_FUEL_BLOCK_HEADER must
+/// be higher than the current block height is met.
+/// The value of the REQUIRED_FUEL_BLOCK_HEADER is set in
+/// the request data by the graphql handler as a value of type
+/// `RequiredHeight`.
+#[derive(Debug, derive_more::Display, derive_more::From)]
+pub(crate) struct RequiredFuelBlockHeightExtension;
+
+impl RequiredFuelBlockHeightExtension {
+    pub fn new() -> Self {
+        Self
+    }
+}
+
+impl ExtensionFactory for RequiredFuelBlockHeightExtension {
+    fn create(&self) -> Arc<dyn Extension> {
+        Arc::new(RequiredFuelBlockHeightExtension::new())
+    }
+}
+
+/// Error value returned by the extension when the required fuel block height
+/// precondition is not met.
+pub(crate) struct RequiredFuelBlockHeightTooFarInTheFuture;
```

Comment:
If this is supposed to be used as an error you might consider implementing `Display` and `std::Error` for it.

Reply:
not an error, it is used upstream to populate the `CURRENT_FUEL_BLOCK_HEIGHT` header in case the precondition is not met.

Reply:
Ok, thanks for explanation. I got confused by the comment above :-)